doc/_utils/redirects.py

```diff
@@ -4,455 +4,474 @@
+    ("getting_started/programming", "config_and_build/programming"),
+    ("config_and_build/programming", "app_dev/programming"),
+    ("config_and_build/companion_components", "app_dev/companion_components"), # Using companion components
+    ("config_and_build/output_build_files", "config_and_build/configuring_app/output_build_files"), # Output build files (image files)
     ("config_and_build/configuring_app/output_build_files", "app_dev/config_and_build/output_build_files"),
-    ("config_and_build/pin_control", "config_and_build/configure_app/hardware/pin_control"),
-    ("config_and_build/use_gpio_pin_directly", "config_and_build/configure_app/hardware/use_gpio_pin_directly"),
-    ("ug_bootloader", "config_and_build/bootloaders_and_dfu/bootloader"),
-    ("app_dev/bootloaders_and_dfu/bootloader", "config_and_build/bootloaders_and_dfu/bootloader"),
-    ("app_dev/bootloaders_and_dfu/bootloader", "config_and_build/bootloaders/bootloader"),
-    ("config_and_build/bootloaders/bootloader", "config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader"),
-    ("config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader", "app_dev/bootloaders_dfu/mcuboot_nsib/bootloader"),
-    ("ug_bootloader_adding", "config_and_build/bootloaders_and_dfu/bootloader_adding"),
+    ("ug_multi_image", "config_and_build/multi_image"), # Multi-image build using child and parent images
+    ("app_dev/multi_image/index", "config_and_build/multi_image"),
+    ("config_and_build/multi_image", "app_dev/config_and_build/multi_image"),
+    ("ug_fw_update", "config_and_build/bootloaders_and_dfu/fw_update"), # Firmware updates (removed after 2.5.0)
+    ("app_dev/bootloaders_and_dfu/fw_update", "config_and_build/bootloaders_and_dfu/fw_update"),
+    ("config_and_build/bootloaders_and_dfu/fw_update", "config_and_build/dfu/index"),
+    ("config_and_build/dfu/index", "app_dev/dfu/index"), # Device Firmware Updates (removed after 2.7.0) > Bootloaders and DFU (landing)
+    ("app_dev/dfu/index", "app_dev/bootloaders_dfu/index"),
+    ("app_bootloaders", "config_and_build/bootloaders_and_dfu/index"), # Bootloaders (removed after 2.7.0) > Bootloaders and DFU (landing)
+    ("app_dev/bootloaders_and_dfu/index", "config_and_build/bootloaders_and_dfu/index"),
+    ("config_and_build/bootloaders_and_dfu/index", "config_and_build/bootloaders_dfu/index"),
+    ("config_and_build/bootloaders_dfu/index", "app_dev/bootloaders_dfu/index"),
+    ("config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_mcuboot_nsib", "app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_mcuboot_nsib"), # MCUboot and NSIB (landing)
+    ("ug_bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"), # Testing the bootloader chain (removed after 2.7.0)
+    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"),
+    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders/bootloader_testing"),
```

Comment:
2 entries for the same source. Not working. Is it supposed to be
```suggestion
    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"),
    ("config_and_build/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders/bootloader_testing"),
```